File: Uyanda.Customer.MicroService/Uyanda.Customer.Application/Features/CustomerManagement/Models/CustomerModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Uyanda.Customer.Application.Features.CustomerManagement.Models
{
    public class CustomerModel
    {
        public int Id { get; set; }

        public string? Name { get; set; }

        public string? Surname { get; set; }

        public int Points { get; set; }

        public int ContactDetailsId { get; set; }
        public ContactDetailsModel? ContactDetails { get; set; }
    }
}

File: Uyanda.GUI.MicroService/src/app/models/customer.model.ts
export interface ContactDetails {
    id: number;
    emailAddress: string | null;
    phoneNumber: string | null;
}

export interface CustomerStatus {
    id: number;
    customerId: number;
    isFound: boolean;
}

export interface Customer {
    id: number;
    name: string | null;
    surname: string | null;
    points: number;
    contactDetailsId: number;
    contactDetails: ContactDetails | null;
    CustomerStatusId: number;
    customerStatus: CustomerStatus | null;
}